static/app/views/performance/content.tsx

```diff
@@ -33,86 +33,92 @@ import Onboarding from './onboarding';
 import {addRoutePerformanceContext, handleTrendsClick} from './utils';
 
 type Props = {
-  api: Client;
-  organization: Organization;
   selection: GlobalSelection;
   location: Location;
   router: InjectedRouter;
-  projects: Project[];
-  loadingProjects: boolean;
   demoMode?: boolean;
 };
 
 type State = {
   eventView: EventView;
-  error: string | undefined;
+  error?: string;
 };
 
-class PerformanceContent extends Component<Props, State> {
-  static getDerivedStateFromProps(nextProps: Readonly<Props>, prevState: State): State {
-    return {
-      ...prevState,
-      eventView: generatePerformanceEventView(
-        nextProps.organization,
-        nextProps.location,
-        nextProps.projects
-      ),
-    };
-  }
+function PerformanceContent({selection, location, demoMode}: Props) {
+  const api = useApi();
+  const organization = useOrganization();
+  const {projects} = useProjects();
+  const previousDateTime = usePrevious(selection.datetime);
```

Comment:
since DateTime has object values, it needs to be deep compared, and one way to achieve that in a function component is using the `usePrevious` hook.  More details:

https://dev.to/vsramalwan/ways-to-handle-deep-object-comparison-in-useeffect-hook-1elm

Reply:
FYI, a simpler way to do comparison for `useEffect` is changing an object to a primitive representation (just transforming date to a date string or timestamp number).